propolis/src/hw/nvme/cmds.rs

```diff
@@ -607,24 +607,17 @@ impl PrpIter<'_> {
                     // entry which should be present in PRP2
                     PrpNext::Prp2
                 } else {
-                    /*
-                     * The first PRP List entry (i.e. the first pointer to a memory page containing
-                     * additional PRP entries) that if present is contained in the PRP Entry 2 location
-                     * within the command, shall be Qword aligned and may also have a non-zero offset within
-                     * the memory page.
-                     */
+                    // The first PRP List entry (i.e. the first pointer to a memory page containing
+                    // additional PRP entries) that if present is contained in the PRP Entry 2
+                    // location within the command, shall be Qword aligned and may also have a
+                    // non-zero offset within the memory page.
                     if (self.prp2 % 8) != 0 {
                         return Err("PRP2 not Qword aligned!");
                     }
 
-                    /*
-                     * If this entry is
-                     *
-                     * - not the first PRP entry in the command, or
-                     * - in the PRP List
-                     *
-                     * then the Offset portion of this field shall be cleared to 0h.
-                     */
+                    // PRP2 is allowed to have a non-zero offset into the memory page, but mask it
+                    // out when creating the base part of the PrpNext::List enum - the offset will
+                    // be passed with idx.
```

Comment:
The phrasing is confusing here too.  "Allowed to have a non-zero offset, but mask it out" seems almost like it's being discarded.  Perhaps make it clear that the list can start at an offset in the page, but must not cross the page boundary, and that any offset is preserved to index into that region.